lib/schedule.c

```diff
@@ -166,3 +118,230 @@ bool lf_check_deadline(void* self, bool invoke_deadline_handler) {
     }
     return false;
 }
+
+trigger_handle_t lf_schedule_trigger(environment_t *env, trigger_t* trigger, interval_t extra_delay, lf_token_t* token) {
```

Comment:
Note to other reviewers: Based on the original code from `reactor_common.c` that got deleted, this is what the diff should actually look like (relative to the target of this PR):
```
diff --git a/v0.txt b/v1.txt
index d459bde..7f7c928 100644
--- a/v0.txt
+++ b/v1.txt
@@ -1,5 +1,5 @@
-    assert(env != GLOBAL_ENVIRONMENT);
-    if (_lf_is_tag_after_stop_tag(env, env->current_tag)) {
+assert(env != GLOBAL_ENVIRONMENT);
+    if (lf_is_tag_after_stop_tag(env, env->current_tag)) {
         // If schedule is called after stop_tag
         // This is a critical condition.
         _lf_done_using(token);
@@ -12,13 +12,13 @@
         extra_delay = 0LL;
     }
 
-    LF_PRINT_DEBUG("_lf_schedule: scheduling trigger %p with delay " PRINTF_TIME " and token %p.",
+    LF_PRINT_DEBUG("lf_schedule_trigger: scheduling trigger %p with delay " PRINTF_TIME " and token %p.",
             trigger, extra_delay, token);
 
     // Increment the reference count of the token.
     if (token != NULL) {
         token->ref_count++;
-        LF_PRINT_DEBUG("_lf_schedule: Incremented ref_count of %p to %zu.",
+        LF_PRINT_DEBUG("lf_schedule_trigger: Incremented ref_count of %p to %zu.",
                 token, token->ref_count);
     }
 
@@ -41,11 +41,11 @@
     }
     tag_t intended_tag = (tag_t){.time = env->current_tag.time + delay, .microstep = 0};
 
-    LF_PRINT_DEBUG("_lf_schedule: env->current_tag.time = " PRINTF_TIME ". Total logical delay = " PRINTF_TIME "",
+    LF_PRINT_DEBUG("lf_schedule_trigger: env->current_tag.time = " PRINTF_TIME ". Total logical delay = " PRINTF_TIME "",
             env->current_tag.time, delay);
     interval_t min_spacing = trigger->period;
 
-    event_t* e = _lf_get_new_event(env);
+    event_t* e = lf_get_new_event(env);
 
     // Initialize the next pointer.
     e->next = NULL;
@@ -101,11 +101,11 @@
                 found = found->next;
                 intended_tag.microstep++;
             }
-            if (_lf_is_tag_after_stop_tag(env, intended_tag)) {
+            if (lf_is_tag_after_stop_tag(env, intended_tag)) {
                 LF_PRINT_DEBUG("Attempt to schedule an event after stop_tag was rejected.");
                 // Scheduling an event will incur a microstep
                 // after the stop tag.
-                _lf_recycle_event(env, e);
+                lf_recycle_event(env, e);
                 return 0;
             }
             // Hook the event into the list.
@@ -134,14 +134,14 @@
                     // Recycle the new event and decrement the
                     // reference count of the token.
                     _lf_done_using(token);
-                    _lf_recycle_event(env, e);
+                    lf_recycle_event(env, e);
                     return(0);
                 case replace:
                     LF_PRINT_DEBUG("Policy is replace. Replacing the previous event.");
                     // If the event with the previous time is still on the event
                     // queue, then replace the token.  To find this event, we have
                     // to construct a dummy event_t struct.
-                    event_t* dummy = _lf_get_new_event(env);
+                    event_t* dummy = lf_get_new_event(env);
                     dummy->next = NULL;
                     dummy->trigger = trigger;
                     dummy->time = trigger->last_tag.time;
@@ -150,13 +150,13 @@
                     if (found != NULL) {
                         // Recycle the existing token and the new event
                         // and update the token of the existing event.
-                        _lf_replace_token(found, token);
-                        _lf_recycle_event(env, e);
-                        _lf_recycle_event(env, dummy);
+                        lf_replace_token(found, token);
+                        lf_recycle_event(env, e);
+                        lf_recycle_event(env, dummy);
                         // Leave the last_tag the same.
                         return(0);
                     }
-                    _lf_recycle_event(env, dummy);
+                    lf_recycle_event(env, dummy);
 
                     // If the preceding event _has_ been handled, then adjust
                     // the tag to defer the event.
@@ -188,11 +188,11 @@
 
     // Do not schedule events if if the event time is past the stop time
     // (current microsteps are checked earlier).
-    LF_PRINT_DEBUG("Comparing event with elapsed time " PRINTF_TIME " against stop time " PRINTF_TIME ".", e->time - start_time, env->stop_tag.time - start_time);
+    LF_PRINT_DEBUG("Comparing event with elapsed time " PRINTF_TIME " against stop time " PRINTF_TIME ".", e->time - lf_time_start(), env->stop_tag.time - lf_time_start());
     if (e->time > env->stop_tag.time) {
-        LF_PRINT_DEBUG("_lf_schedule: event time is past the timeout. Discarding event.");
+        LF_PRINT_DEBUG("lf_schedule_trigger: event time is past the timeout. Discarding event.");
         _lf_done_using(token);
-        _lf_recycle_event(env, e);
+        lf_recycle_event(env, e);
         return(0);
     }
 
@@ -208,7 +208,7 @@
     // rather than the event_q, so anything put in the event_q with this
     // same time will automatically be executed at the next microstep.
     LF_PRINT_LOG("Inserting event in the event queue with elapsed time " PRINTF_TIME ".",
-            e->time - start_time);
+            e->time - lf_time_start());
     pqueue_insert(env->event_q, e);
 
     tracepoint_schedule(env->trace, trigger, e->time - env->current_tag.time);
```